app/controllers/health_cards_controller.rb

```diff
@@ -43,4 +47,22 @@ def health_card
   def find_patient
     @patient = Patient.find(params[:patient_id])
   end
+
+  def create_immunizations(bundle)
+    immunizations = bundle.entry.select { |e| e.resource.is_a?(FHIR::Immunization) }
+    vaccine_id_map = {
+      '207': 1,
+      '208': 2,
+      '212': 3
+    }
+
+    immunizations.each do |i|
+      immunization_resource = i.resource
+      @pat.immunizations.new({
+                               vaccine_id: vaccine_id_map[immunization_resource.vaccineCode.coding[0].code.to_sym],
```

Comment:
Similarly will there be errors if these elements aren't present? I know they are required by the FHIR IG, but for general Smart Health Cards we might want to only initialize an immunization if those data minimization elements are present in the bundle that is passed in.

Reply:
I think the `vaccineCode` and `occurenceDateTime` have to be in the immunization. I think it will just an empty string if `lotNumber` is not present.